k8s/provider.go

```diff
@@ -93,6 +96,12 @@ func (p *ServiceProvider) Get(ctx context.Context) ([]consul.ServiceInstance, er
 	// TODO(medzin): Allow to specify which containers and ports will be registered
 	container := pod.Spec.Containers[0]
 	host := pod.GetStatus().GetPodIP()
```

Comment:
We can rely only on `getIPRetry()` and remove this call as well as`if` statement below, because they are redundant.

Reply:
If the value is there, than there is no point in having the client re-get the whole structure. But I think what you suggested earlier (check the pod at initial get) is much cleaner.

Reply:
Your function returns as soon as there is a correct pod info, so there is no need, to pre-query the api.